edge/yaml/yaml.go

```diff
@@ -152,6 +165,19 @@ func (y *yaml) AddImagePullSecrets() (string, error) {
 	return strings.Join(ymlFiles, "---\n"), nil
 }
 
+func (y *yaml) AddCredentialsAsEnvForSpecificService(serviceName string) (string, error) {
```

Comment:
I think it's a bit confusing that we have docker and kubernetes in the same file and the same type.  I suggest moving the new logic to `yaml/docker.go` and creating a new type for this `dockerYaml`